modules/javafx.graphics/src/test/java/test/javafx/scene/layout/BorderPaneTest.java

```diff
@@ -915,6 +923,56 @@ public class BorderPaneTest {
         assertEquals(5, center.getLayoutY(), 1e-100);
     }
 
+    @CsvSource({
+            "209.0, 430.0, false, false", // flowPane 4x1, no margin, no padding
+            "211.0, 210.0, false, false", // flowPane 2x2, no margin, no padding
+            "209.0, 430.0, true, false",  // flowPane 4x1, margin, no padding
+            "211.0, 210.0, true, false",  // flowPane 2x2, margin, no padding
+            "209.0, 430.0, false, true",  // flowPane 4x1, no margin, padding
+            "211.0, 210.0, false, true",  // flowPane 2x2, no margin, padding
+            "209.0, 430.0, true, true",   // flowPane 4x1, margin, padding
+            "241.0, 210.0, true, true",   // flowPane 2x2, margin, padding
+    })
+    @ParameterizedTest
+    public void testFlowPaneCenterChildWithPaddingAndMargin(double width, double minHeight, boolean useMargin, boolean usePadding) {
+        FlowPane center = new FlowPane(10, 10);
```

Comment:
I wonder if we should also test the layout-related code using fractional scale + snapping.
what do you think?